src/context/AppProvider/AppContext.tsx

```diff
@@ -171,6 +174,9 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
       // and ensure the queryFn runs resulting in dispatches occuring to update client state
       const options = { forceRefetch: true }
 
+      if (featureFlags.DynamicLPAssets) {
+        await dispatch(zapperApi.endpoints.getZapperUniV2PoolAssetIds.initiate())
```

Comment:
`await`ing the thunk looks like it's unsafe and could actually throw but actually *is* safe, see https://redux-toolkit.js.org/api/createAsyncThunk#unwrapping-result-actions. All this does is wait for the thunk to be *settled* (not fulfilled), error-handling is then done in a monadic way:

> The thunks generated by createAsyncThunk will always return a resolved promise with either the fulfilled action object or rejected action object inside, as appropriate.